### PR TITLE
KT-33594 "Code Cleanup" removes import of @Deprecated ReplaceWith declaration with suppressed deprecations

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-33594

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/KotlinCleanupInspection.kt [hunk 1]
```diff
@@ -70,6 +76,17 @@ class KotlinCleanupInspection : LocalInspectionTool(), CleanupLocalInspectionToo
         return problemDescriptors.toTypedArray()
     }
 
+    private fun KtFile.hasSuppressDeprecationAnnotation(): Boolean {
+        val suppressAnnotationEntry = annotationEntries.firstOrNull {
+            it.shortName?.asString() == "Suppress"
+                    && it.resolveToCall()?.resultingDescriptor?.containingDeclaration?.fqNameSafe == KotlinBuiltIns.FQ_NAMES.suppress
+        } ?: return false
+        return suppressAnnotationEntry.valueArguments.any {
+            val text = (it.getArgumentExpression() as? KtStringTemplateExpression)?.entries?.firstOrNull()?.text ?: return@any false
```
Comment (nikitabobko) [inline idea/src/org/jetbrains/kotlin/idea/inspections/KotlinCleanupInspection.kt:85]: > firstOrNull()

"singeOrNull" seems safer option for me

File: idea/tests/org/jetbrains/kotlin/idea/inspections/KotlinCleanupInspectionTest.kt [hunk 2]
```diff
@@ -35,10 +35,14 @@ class KotlinCleanupInspectionTest() : KotlinLightCodeInsightFixtureTestCase() {
         val profile = InspectionProjectProfileManager.getInstance(project).currentProfile
         globalContext.codeCleanup(analysisScope, profile, "Cleanup", null, true)
 
-        myFixture.checkResultByFile(result)
+        myFixture.checkResultByFile("$dir/$result")
     }
 
-    fun testCleanup() {
-        doTest("cleanup.kt.after", "cleanup.kt", "JavaAnn.java", "deprecatedSymbols.kt")
+    fun testBasic() {
```
Comment (nikitabobko) [inline idea/tests/org/jetbrains/kotlin/idea/inspections/KotlinCleanupInspectionTest.kt:41]: You renamed `testCleanup` to `testBasic` but left `cleanup.kt` and `cleanup.kt.after` names. I guess they should be renamed to `basic.kt` and `basic.kt.after` as well

File: idea/tests/org/jetbrains/kotlin/idea/inspections/KotlinCleanupInspectionTest.kt [hunk 3]
```diff
@@ -35,10 +35,14 @@ class KotlinCleanupInspectionTest() : KotlinLightCodeInsightFixtureTestCase() {
...
+        doTest("basic", "cleanup.kt.after", "cleanup.kt", "JavaAnn.java", "deprecatedSymbols.kt")
+    }
+
+    fun testFileWithAnnotationToSuppressDeprecation() {
+        doTest("fileWithAnnotationToSuppressDeprecation", "foo.kt.after", "foo.kt", "bar.kt")
```
Comment (nikitabobko) [inline idea/tests/org/jetbrains/kotlin/idea/inspections/KotlinCleanupInspectionTest.kt:46]: Same here. `foo.kt` and `foo.kt.after` better to rename to something meaningful

**Comment by yanex** [comment]
Review: OK.

**Comment by nikitabobko** [comment]
Merged to new repo https://github.com/jetbrains/intellij-kotlin/commit/b8c1e87d65f1d79b1e1b3724eda7cfb1bab85114

**Pull request closed without merging**